### PR TITLE
Implement 'run_uuid' handling in mlflow/server/auth/__init__.py

### LINKED ISSUE #8511: [BUG] No 'run_uuid' handling while validating user permissions
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

- Client: 2.3.2
- Tracking server: 2.3.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Python version**:
- **yarn version, if running the dev UI**:

### Describe the problem

When login to the mlflow server as a non-admin user, mlflow checks user permissions before displaying run information ([here](https://github.com/gabrielfu/mlflow/blob/7490af79397922263c9580fd18fcc66daf8ffe18/mlflow/server/auth/__init__.py#L406)). It is implemented with use of `run_id` ([here](https://github.com/gabrielfu/mlflow/blob/7490af79397922263c9580fd18fcc66daf8ffe18/mlflow/server/auth/__init__.py#L249)). In case of the artifacts list, mlflow itself creates a response with `run_uuid` instead of `run_id`:

GET | http://localhost:5000/ajax-api/2.0/mlflow/artifacts/list?run_uuid=0707a3949a80428283cf9a39e62b3a46

As a result, user permissions validation fails [here](https://github.com/gabrielfu/mlflow/blob/7490af79397922263c9580fd18fcc66daf8ffe18/mlflow/server/auth/__init__.py#L208), because of `"error_code": "INVALID_PARAMETER_VALUE", "message": "Missing value for required parameter 'run_id'." `
So, non-admin users see this:
image

Exactly the same problem occurs with metric history (it loads forever):
image

I think it is necessary to add `user_uuid` request parameter handling [here](https://github.com/gabrielfu/mlflow/blob/7490af79397922263c9580fd18fcc66daf8ffe18/mlflow/server/auth/__init__.py#L208) (something like [this](https://github.com/mlflow/mlflow/blob/2eba11c608db9c570e20cb0457c2928dcfc56f0a/mlflow/server/handlers.py#L543) or just try-catch block) or [here](https://github.com/gabrielfu/mlflow/blob/7490af79397922263c9580fd18fcc66daf8ffe18/mlflow/server/auth/__init__.py#L151) (something like [this](https://github.com/mlflow/mlflow/blob/2eba11c608db9c570e20cb0457c2928dcfc56f0a/mlflow/server/handlers.py#L438)).

With admin users everything works fine.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
REPLACE_ME
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by ka1mar**
Signed-off-by: Marina Frolova <m.f.26@mail.ru>

## Related Issues/PRs 

Resolve #8511

## What changes are proposed in this pull request?

Added 'run_uuid' check if there is no 'run_id' in the request:
```
if param == "run_id" and "run_uuid" in args:
  return args["run_uuid"]
```

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

Restart the server - now artifacts and metrics history are visible.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [x] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Thanks for the PR! cc @gabrielfu

**Comment by gabrielfu** [comment]
Thanks for reporting this issue! Though I would suggest not hardcoding this logic inside `_get_request_param()`. How about something like this?

```python
def _get_request_param(param: str, optional: bool = False, default = None):
    ...
    if param not in args and not optional:
        raise ...
    return args.get(param, default)


def _get_permission_from_run_id():
    ...
    run_id = _get_request_param("run_id", optional=True) or _get_request_param("run_uuid")
```

**Comment by ka1mar** [comment]
> Thanks for reporting this issue! Though I would suggest not hardcoding this logic inside `_get_request_param()`. How about something like this?
> 
> ```python
> def _get_request_param(param: str, optional: bool = False, default = None):
>     ...
>     if param not in args and not optional:
>         raise ...
>     return args.get(param, default)
> 
> 
> def _get_permission_from_run_id():
>     ...
>     run_id = _get_request_param("run_id", optional=True) or _get_request_param("run_uuid")
> ```

Thanks, I corrected.

**Comment by harupy** [comment]
@ka1mar Thanks for the PR! Can we check `run_uuid` as well if `run_id` is passed as follows?

```
def _get_request_param(param: str):
    ...
    if param not in args and not optional:
        if param == "run_id":
            return _get_request_param("run_uuid")
        raise ...
    return args.get(param, default)
```

**Comment by harupy** [comment]
@ka1mar Thanks for the comment! If we had more parameters like `run_id`, I'd use what @gabrielfu suggested. For now, the parameter that requires special handling is just `run_id`.

**Comment by ka1mar** [comment]
> @ka1mar Thanks for the update! Can we resolve conflicts?

@harupy, done

File: mlflow/server/auth/__init__.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def make_forbidden_response() -> Response:
     return res
 
 
-def _get_request_param(param: str) -> str:
+def _get_request_param(param: str, optional: bool = False, default=None) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (ka1mar) [reply mlflow/server/auth/__init__.py]: Deleted

**Review by harupy — APPROVED** [review]
Looks good to me!

**Comment by harupy** [comment]
@ka1mar Thanks for the PR :)

**Pull request merged**